plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1008,6 +1013,36 @@ public boolean configure(final String name, final Map<String, Object> params) th
         if (_localStoragePath == null) {
             _localStoragePath = "/var/lib/libvirt/images/";
         }
+        _localStorageUUID = (String)params.get("local.storage.uuid");
+        if (_localStorageUUID == null) {
+            _localStorageUUID = UUID.randomUUID().toString();
+        }
+
+        String[] localStorageRelativePaths = _localStoragePath.split(CONFIG_VALUES_SEPARATOR);
+        String[] localStorageUUIDs = _localStorageUUID.split(CONFIG_VALUES_SEPARATOR, -1);
+        if (localStorageRelativePaths.length != localStorageUUIDs.length) {
+            throw new ConfigurationException("The path and UUID of local storage pools have different length");
+        }
+        for (String localStorageRelativePath : localStorageRelativePaths) {
+            final File storagePath = new File(localStorageRelativePath);
+            _localStoragePaths.add(storagePath.getAbsolutePath());
+        }
+        _localStoragePath = StringUtils.join(_localStoragePaths, CONFIG_VALUES_SEPARATOR);
+        params.put("local.storage.path", _localStoragePath);
+
+        for (String localStorageUUID : localStorageUUIDs) {
```

Comment:
If for example: localStoragePath contains 3 paths and localStorageUUIDs contains 2 UUIDS, should we always fail or try to map the maximum pairs? In this example: mapping the first 2 paths with the 2 UUIDs?

Reply:
@nvazquez 
I would prefer to stop , ask user to fix the misconfiguration and retry.
otherwise they may have to dig into agent.log and figure out why the last pool does not work.